src/Steam.Common/App.cs

```diff
@@ -0,0 +1,110 @@
+using System;
+
+namespace Steam
+{
+    /// <summary>
+    /// Main representation of a product, see https://partner.steamgames.com/doc/store/application
+    /// </summary>
+    public readonly struct App : IEquatable<App>
+    {
+        /// <summary>
+        /// The Id
+        /// </summary>
+        public readonly uint Id;
+
+        /// <summary>
+        /// Invalid app value
+        /// </summary>
+        public static readonly App Invalid = 0x0;
+
+        /// <summary>
+        /// Creates a new instance of <see cref="App"/>
+        /// </summary>
+        /// <param name="id">The id</param>
+        public App(uint id)
```

Comment:
Looking further into the App ID, it turns out it's a part of a greater structure called the Game ID (which will be added through #4). Since app ID is a part of game ID, I think it should have a range like that of the app ID section of the game ID which is 24 bits.